### PR TITLE
Add support for X-Remap header to xdebug plugin.

### PR DESCRIPTION
**Pull request by ywkaras**
If a request with the header 'x-debug: x-remap' is sent, and the URL is remapped by the core, then a header like for example:

X-Remap: from=http://three[0-9]+/, to=http://127.0.0.1:666/

will be substituted for x-debug.  The *from* and *to* will be those from the corresponding line in remap.config that caused the URL remapping.

### CODE REVIEW

**Comment by ywkaras** [comment]
[autest ci ubuntu]

File: plugins/xdebug/xdebug.cc [hunk 1]
```diff
@@ -242,6 +243,63 @@ InjectMilestonesHeader(TSHttpTxn txn, TSMBuffer buffer, TSMLoc hdr)
...
+{
+  TSMLoc dst = FindOrMakeHdrField(buffer, hdr, "X-Remap", lengthof("X-Remap"));
+
+  if (TS_NULL_MLOC != dst) {
+    int fromUrlStrLen, toUrlStrLen;
+    const char *fromUrlStr = getRemapUrlStr(txn, TSRemapFromUrlGet, fromUrlStrLen);
+    const char *toUrlStr   = getRemapUrlStr(txn, TSRemapToUrlGet, toUrlStrLen);
+
+    char buf[2048];
+    int len = snprintf(buf, sizeof(buf) - 1, "from=%*s, to=%*s", fromUrlStrLen, fromUrlStr, toUrlStrLen, toUrlStr);
```
Comment (bryancall) [inline plugins/xdebug/xdebug.cc]: Should be sizeof(buf):
     The snprintf() and vsnprintf() functions will write at most size-1 of the characters printed into the output string (the size'th
     character then gets the terminating `\0'); if the return value is greater than or equal to the size argument, the string was too
     short and some of the printed characters were discarded.  The output is always null-terminated, unless size is 0.

**Comment by bryancall** [comment]
Can we get more time on the PRs for other people to review that aren't in the same group.  Also, PRs should be correctly assigned, labeled, and have milestones set.

**Comment by bryancall** [comment]
@ywkaras Should this need to go into 7.2.x?

File: plugins/xdebug/xdebug.cc (hunk 1 above)
Comment (ywkaras) [reply plugins/xdebug/xdebug.cc]: That could be considered a separate issue from this PR, as I followed the pattern I saw in the existing function InjectTxnUuidHeader() .

**Comment by ywkaras** [comment]
It definitely needs to go to Oath internal 7.2.x so presumably that means it should go to Apache TS 7.2.x?

**Pull request merged**